packages/terra-menu/src/Menu.jsx

```diff
@@ -113,11 +114,31 @@ class Menu extends React.Component {
   push(item) {
     this.setState((prevState) => {
       const newStack = prevState.stack.slice();
-      newStack.push(item);
+      let updatedStack = {};
+      if (newStack.length - 1 >= 1) {
+        updatedStack = newStack[0].props.children.filter((list) => list.props.subMenuItems && !list.props.isDisabled);
+        updatedStack = this.recursiveFunction(updatedStack, item.key);
+      } else updatedStack = item;
+      newStack.push(updatedStack);
       return { stack: newStack };
     });
   }
 
+  recursiveFunction(updatedStack, key) {
+    for (let i = 0; i < updatedStack.length; i += 1) {
+      if (updatedStack[i].key === key) {
+        return updatedStack[i];
+      }
+      if (updatedStack[i].props.subMenuItems) {
+        const result = this.recursiveFunction(updatedStack[i].props.subMenuItems, key);
+        if (result) {
+          return result;
+        }
+      }
+    }
+    return null;
+  }
```

Comment:
Can we rename this function with meaning-full Name and lets move this function to [menu-utils](https://github.com/cerner/terra-framework/blob/3d1af6a16f73ba6e05c28ca772219734395f1131/packages/terra-menu/src/_MenuUtils.js)

Can we simplify the above function like below ?
```suggestion
  const findMenuItem = (updatedStack, key) => {
  updatedStack.forEach((menuItem) => {
    if (menuItem.key === key) {
      return menuItem;
    }

    if (menuItem.props.subMenuItems) {
      findMenuItem(menuItem.props.subMenuItems, key);
    }
    return undefined;
  });
};
```